src/sec_edgar_bulker/core/idx_downloader.py

```
@@ -0,0 +1,470 @@
+    
+    def __init__(self, config: Config) -> None:
+        """Initialize the IDX downloader.
+        
+        Args:
+            config: Configuration object
+        """
+        self.config = config
+        self.logger = logging.getLogger("sec_edgar_bulker.idx_downloader")
+        self.header_generator = HeaderGenerator(config)
+        self.proxy_manager = ProxyManager(config) if config.proxy.enabled else None
+        self.temp_files: Dict[str, Path] = {}  # Track temp files for cleanup
```

Ruff (0.8.2)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)